cocotb/share/makefiles/simulators/Makefile.verilator

```diff
@@ -0,0 +1,110 @@
+###############################################################################
+# Copyright (c) 2013 Potential Ventures Ltd
+# Copyright (c) 2013 SolarFlare Communications Inc
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#     * Neither the name of Potential Ventures Ltd,
+#       SolarFlare Communications Inc nor the
+#       names of its contributors may be used to endorse or promote products
+#       derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL POTENTIAL VENTURES LTD BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+
+ifneq ($(VHDL_SOURCES),)
+
+results.xml:
+	@echo "Skipping simulation as VHDL is not supported on simulator=$(SIM)"
+debug: results.xml
+clean::
+
+else
+
+CMD := verilator
+
+ifeq ($(shell which $(CMD) 2>/dev/null),)
+# Verilator is not in PATH, lets start searching for it
+  ifdef VERILATOR_ROOT
+    CMD = $(VERILATOR_ROOT)/bin/verilator
+  else
+    PKG_CONFIG ?= pkg-config
+    ifneq ($(shell which $(PKG_CONFIG) 2>/dev/null),)
+      # Use pkg-config to find it
+      EXEC = $(shell $(PKG_CONFIG) --variable=exec_prefix verilator)
+      ifeq ($(EXEC),)
+        $(error Cannot find verilator, either set VERILATOR_ROOT or add it to PKG_CONFIG_PATH.)
+      endif
+      CMD = $(EXEC)/bin/verilator
+    else
+        $(error Cannot find verilator, either set VERILATOR_ROOT or add it to PKG_CONFIG_PATH.)
+    endif
+  endif
+endif
+
+ifeq ($(VERILATOR_SIM_DEBUG), 1)
+  COMPILE_ARGS += --debug
+  PLUSARGS += +verilator+debug
+  SIM_BUILD_FLAGS += -DVL_DEBUG
+endif
+
+ifeq ($(VERILATOR_TRACE),1)
+  EXTRA_ARGS += --trace
+endif
+
+ifdef PRECISION
```

Comment:
Please call this ``COCOTB_HDL_TIMEPRECISION`` so that it will automatically work with #1113.